digits/scheduler.py

```diff
@@ -175,6 +177,30 @@ def add_job(self, job):
             return False
         else:
             self.jobs.append(job)
+
+            if flask._app_ctx_stack.top is not None:
```

Comment:
Not quite. New error. Some new caffe errors, so rebuilt caffe and pycaffe.  Incidentally make -j10 doesn't work in caffe until the first few targets build.  Looking into the new error.

======================================================================
ERROR: Failure: AttributeError ('module' object has no attribute 'scheduler')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/local/lib/python2.7/dist-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/local/lib/python2.7/dist-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/jmancewicz/dev/digits/DIGITS/digits/test_scheduler.py", line 7, in <module>
    from . import scheduler as _
  File "/home/jmancewicz/dev/digits/DIGITS/digits/scheduler.py", line 23, in <module>
    from digits.webapp import app
  File "/home/jmancewicz/dev/digits/DIGITS/digits/webapp.py", line 20, in <module>
    scheduler = digits.scheduler.Scheduler(config_value('gpu_list'))
AttributeError: 'module' object has no attribute 'scheduler'